stackslib/src/net/api/postblock_proposal.rs

```diff
@@ -75,6 +76,21 @@ pub enum ValidateRejectCode {
     UnknownParent,
 }
 
+impl TryFrom<u8> for ValidateRejectCode {
+    type Error = String;
+
+    fn try_from(value: u8) -> Result<Self, Self::Error> {
+        match value {
+            0 => Ok(ValidateRejectCode::BadBlockHash),
+            1 => Ok(ValidateRejectCode::BadTransaction),
+            2 => Ok(ValidateRejectCode::InvalidBlock),
+            3 => Ok(ValidateRejectCode::ChainstateError),
+            4 => Ok(ValidateRejectCode::UnknownParent),
+            _ => Err(format!("Invalid value for ValidateRejectCode: {value}")),
+        }
+    }
+}
```

Comment:
Related comment about `define_u8_enum!` here -- if you use that to define the enum, you could implement this without a potentially dangerous match.